src/emulator/commandUtils.ts

```diff
@@ -156,13 +169,59 @@ export function parseInspectionPort(options: any): number {
   return parsed;
 }
 
-export function shutdownWhenKilled(): Promise<void> {
+/**
+ * Sets the correct export options based on --import and --export-on-exit.
+ * Also validates if we have a correct setting we need to export the data on exit.
+ * When used as: `--import ./data --export-on-exit` or `--import ./data --export-on-exit ./data`
+ * we do allow an non-existing --import [dir] and we just export-on-exit. This because else one would always need to
+ * export data the first time they start developing on a clean project.
+ * @param options
+ */
+export function setExportOnExitOptions(options: any): any {
+  if (options.exportOnExit) {
+    if (typeof options.import === "string") {
```

Comment:
You can assume everything in `options` (or at least all the flags) are strings when they're defined, no need to do all this `typeof` checking.

If you're just using `typeof` to check if it exists, we prefer checking the truthiness instead.  So for example you have this:

```js
options.exportOnExit =
        typeof options.exportOnExit === "string" ? options.exportOnExit : options.import;
```

I think you can just do:
```js
options.exportOnExit = options.exportOnExit || options.import;
```

Reply:
`options.exportOnExit` can be a boolean when used without a [dir]. Like `--import ./data --export-on-exit`

Reply:
Ah yes good point!  Want to just add a comment about this?  Also seems like `typeof options.import` checks are still not necessary.

Reply:
Agree it is not the prettiest, added some test cases and a note.